File: 1379_Find_A_Corresponding_Node_Of_A_Binary_Tree_In_A_Cloned_Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def getTargetCopy(self, original: TreeNode, cloned: TreeNode, target: TreeNode) -> TreeNode:
        if original == None:
            return None    
        queue = []
        queue.append(cloned) 
        while queue: 
            n = len(queue) 
            root = queue.pop() 
            if root.val == target.val:
                return root
            if root.left != None:
                queue.append(root.left)
            if root.right != None:
                queue.append(root.right) 
        return None 
                
                
        

File: 1379_Find_A_Corresponding_Node_Of_A_Binary_Tree_In_A_Cloned_Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */

class Solution {
public:
    TreeNode* getTargetCopy(TreeNode* original, TreeNode* cloned, TreeNode* target) {
        if(original == NULL || cloned == NULL) return NULL; 
        vector<TreeNode*> queue; 
        queue.push_back(cloned); 
        while(queue.size() > 0){
            TreeNode* current = queue[0];
            queue.erase(queue.begin());
            if(current->val == target->val){ return current; }
            if(current->left != NULL){ queue.push_back(current->left);}
            if(current->right != NULL){ queue.push_back(current->right); }
        }
        return NULL; 
    }
};